src/worker/tester.py

```diff
@@ -40,21 +46,39 @@ def _init_output(self):
 
     def _update_output(self, epoch_output, products):
         """ Update the dictionary saver: extend entries """
-        data, model_output = products['data'], products['model_output']
 
-        def fetch_from_dict(dictionary):
+        def update_epoch_output_from_dict(dictionary):
             for key in dictionary.keys():
                 if key not in global_config.saved_keys:
                     continue
                 value = dictionary[key]
                 saved_value = value.cpu().numpy() if torch.is_tensor(value) else value
                 epoch_output['saved'][key].extend([v for v in saved_value])
 
-        fetch_from_dict(data)
-        fetch_from_dict(model_output)
+        data, model_output = products['data'], products['model_output']
+        if global_config.save_while_infer:
+            # Clean previous results
+            epoch_output['saved'] = {k: [] for k in global_config.saved_keys}
+
+        for d in [data, model_output]:
+            update_epoch_output_from_dict(d)
+
+        if global_config.save_while_infer:
+            # Save results
+            name = self.data_loader.name
+            for i, output in enumerate(epoch_output['saved']['model_output']):
+                index = epoch_output['saved']['index'][i]
+                output_path = os.path.join(self.saving_dir, f'{name}_index{index:06d}.npy')
+                np.save(output_path, output)
+                if index % 1000 == 0:
+                    logger.info(f'Saving output {output_path}...')
```

Comment:
This output message will show up asynchronously with that of `_print_log`?

Reply:
_print_log will only show the current batch num